src/gui/dialogues/newpilotdialog.cpp

```diff
@@ -65,8 +65,12 @@ NewPilotDialog::NewPilotDialog(QWidget *parent) :
     setupCompleter();
 
     using namespace experimental;
-    //    connect(DB(), &DataBase::commitUnsuccessful,
-    //            this, &NewPilotDialog::onCommitUnsuccessful);
+
+    QObject::connect(DB(), &DataBase::commitSuccessful,
+            this, &NewPilotDialog::onCommitSuccessful);
+    QObject::connect(DB(), &DataBase::commitUnsuccessful,
+            this, &NewPilotDialog::onCommitUnsuccessful);
```

Comment:
Maybe do the connection in a higher level. Instead of the dialog re-connecting itself. Have the `mainwindow` connect `on_newPilot_triggered`?